packages/edit-post/src/components/header/header-toolbar/index.js

```diff
@@ -65,23 +68,50 @@ function HeaderToolbar( { onToggleInserter, isInserterOpen } ) {
 			className="edit-post-header-toolbar"
 			aria-label={ toolbarAriaLabel }
 		>
-			<Button
-				className="edit-post-header-toolbar__inserter-toggle"
-				isPrimary
-				isPressed={ isInserterOpen }
-				onClick={ onToggleInserter }
-				disabled={ ! isInserterEnabled }
-				icon={ plus }
-				label={ _x(
-					'Add block',
-					'Generic label for block inserter button'
+			<ToolbarItem>
+				{ ( itemProps ) => (
+					<Button
+						{ ...itemProps }
+						className="edit-post-header-toolbar__inserter-toggle"
+						isPrimary
+						isPressed={ isInserterOpen }
+						onClick={ onToggleInserter }
+						disabled={ ! isInserterEnabled }
+						icon={ plus }
+						label={ _x(
+							'Add block',
+							'Generic label for block inserter button'
+						) }
+					/>
+				) }
+			</ToolbarItem>
+			{ isLargeViewport && (
+				<ToolbarItem>
+					{ ( itemProps ) => <ToolSelector { ...itemProps } /> }
+				</ToolbarItem>
+			) }
+			<ToolbarItem>
+				{ ( itemProps ) => <EditorHistoryUndo { ...itemProps } /> }
+			</ToolbarItem>
```

Comment:
Here and it other similar places, could it be coded as:
```suggestion
			<ToolbarItem as={ EditorHistoryUndo } />
```

I like this shorter version and guess who did I learn about it from? 😄

Reply:
Yeah! I agree with you. Just added support for the `as` prop on `ToolbarItem`.

Reply:
I assumed it was there already but apparently backward compatibility made it harder 😃